### DISCUSSION TITLE
How to support more Android devices?

### DISCUSSION THREAD

**Discussion by KupoGames** [airsdk/Adobe-Runtime-Support#3253] [General]
UPDATE:
Adding this to the XML seems to have done the trick!
`<buildArchitectures>armv7,armv8,x86,x64</buildArchitectures>`
Google now says that 19,963 devices are supported.

-------------------------------------------------------------------------

Hey, I'm wondering if anyone knowledgable in Android development can shed some light on this issue.

About 2 years ago I published my app "Epic Battle Fantasy 5" on Google Play, and have not kept it up to date - the device catalog says that around 20K devices are supported.
The app uses AIR 33 and Distriqt's Ads, In-App purchases, App Rater, Cloud Saving and Achievements ANEs.
In the manifest file I have: android:minSdkVersion="19" android:targetSdkVersion="31"

Now I published "Epic Battle Fantasy 4", using updated ANEs and AIR 50, and the device catalog says that only 9.5K devices are supported.
In the manifest file I have: android:minSdkVersion="19" android:targetSdkVersion="33"

Have I misconfigured something? The compatibility doesn't appear to rely on the Android version, some users are suggesting that maybe 32bit devices aren't working? 

What are the best practices for supporting the most Android devices?

Thanks.

**Comment by lwmirkk** [comment]
Can be some application funcionality?
uses-permission, uses-feature

**Comment by FliplineStudios** [comment]
That's odd, I just checked one of our older apps (using 33.1.1.935) and our newest (using 50.2.4.4) and they have the same number of supported device models (19,131 in our case).

How are you packaging the app, and which version of AIR 50.x are you using?  I think there was an issue with anything < 50.2.4.3 where the 32-bit libraries weren't always included in AABs, not sure if that was Animate-specific.  You can also unzip your AAB and make sure you see an armeabi-v7a folder in base/lib to confirm the 32-bit libraries are there.

**Reply by KupoGames** [reply]
Oh wow, I didn't know AAB files could be unzipped so simply, haha. That's useful.
You're exactly right - my old app has 4 folders in there: arm64-v8a, armeabi-v7a, x86, and x86_64
My new app only has one folder: arm64-v8a

I'm using AIR 50.2.4.5, and compiling from Flash CC 2015.
As far as I can tell, my compile settings are all the same as before.
I edit the adt.cfg file to make sure these lines are there:
DefaultArch=armv8
OverrideArch=armv8
CreateAndroidAppBundle=true

I guess I can try a different version of AIR or play around with some other settings.

**Reply by ajwfrost** [reply]
I would suggest commenting the lines:
```
DefaultArch=armv8
OverrideArch=armv8
```
because that would mean it tries to just build the armv8 version.
Mind you, if it's from Animate (or Flash CC) then it will be passing in an "-armv7" command line argument.. 
One thing you could try is adding to your application descriptor XML file, in the `<android>` section:
```
<buildArchitectures>armv7,armv8,x86,x64</buildArchitectures>
```

See https://airsdk.dev/docs/building/application-descriptor-files/elements/android#buildarchitectures

**Reply by KupoGames** [reply]
Adding this to the XML seems to have done the trick, thanks!
`<buildArchitectures>armv7,armv8,x86,x64</buildArchitectures>`

Google now says that 19,963 devices are supported. Very cool.

**Comment by cgrobb** [comment]
I'm attempting to understand why a [Lenovo Chromebook Duet 3 (11”)](https://www.lenovo.com/us/en/p/laptops/lenovo/lenovo-edu-chromebooks/ideapad-duet-3-chromebook-11-inch,-qlc/len101i0034) isn't supported by the Play Store (and is not in the Device catalog). Issue is detailed over in Play Console land.

https://support.google.com/googleplay/android-developer/thread/289455810?hl=en

Any advice is appreciated.

**Reply by cgrobb** [reply]
Update: This was actually a silent filtering by a parental control that was active--this was my child's device.
